spring-cloud-kubernetes-client-config/src/main/java/org/springframework/cloud/kubernetes/client/config/KubernetesClientSecretsPropertySource.java

```diff
@@ -99,11 +100,17 @@ private static String createLabelsSelector(Map<String, String> labels) {
 		return selectorString.toString();
 	}
 
-	private static void putAll(V1Secret secret, Map<String, Object> result) {
+	private static void putAll(V1Secret secret, Map<String, Object> result, boolean useNameAsPrefix) {
 		Map<String, String> secretData = new HashMap<>();
 		secret.getData().forEach((key, value) -> secretData.put(key, Base64.getEncoder().encodeToString(value)));
 		if (secret != null) {
-			putAll(secretData, result);
+
+			String secretName = null;
+			if (secret.getMetadata() != null) {
+				secretName = secret.getMetadata().getName();
+			}
+
+			putAll(secretData, result, secretName, useNameAsPrefix);
```

Comment:
I agree with @wind57 on not needing to pass `useNameAsPrefix` and doing that check here